### ISSUE TITLE
Does not warn about unquoted $?

### ISSUE THREAD

**Issue by a1346054** [koalaman/shellcheck#2341]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2181
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

1st example:
```sh
#!/bin/sh
IFS=0
true
if [ $? = 0 ]; then
    echo "foo"
else
    echo "bar"
fi
```

2nd example:
```sh
#!/bin/sh
IFS=0
true
echo returned code was $?
```

#### Here's what shellcheck currently says:

For the 1st example:
```
$ shellcheck myscript
 
Line 4:
if [ $? = 0 ]; then
     ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
```

For the 2nd example:
```
$ shellcheck myscript
No issues detected!
```

#### Here's what I wanted or expected to see:

Sometimes scripts are run in a way where the IFS is not set to an expected value or the developer has no control over it

Lots of people ignore SC2181 as a style choice for the 1st example. In both examples, shellcheck should issue SC2086 because the only way to make that script work is to double quote the "$?".

The wiki page for SC2181 or SC2086 does not cover this case.

Possibly similar issues exist for $$ and other vars like that.

**Comment by koalaman** [comment]
You're right, this is one of the long tail of things that can go wrong with a script. It could also be that `sh` is `bash` and that it picks up definitions of `[` or `echo` from the environment, or that `if` is an alias, and even if all those are carefully unset,(taking care not to invoke a potentially rigged `unalias` command) there could be a DEBUG trap that resets them immediately. And what if the shell has custom builtins loaded?

It would be fun to try to make scripts bulletproof from all of the above, but it's out of scope for ShellCheck which instead tries to pragmatically find likely sources of realistic, unintentional issues like `IFS='\0'`.
